### PR TITLE
Preserve list type query paramaters when merging QueryParams objects

### PR DESCRIPTION
**Pull request by kcen**
Addresses: https://github.com/encode/httpx/issues/546

Attempting to address the issue, whilst keeping the element of surprise low for existing use cases.

### CODE REVIEW

File: httpx/models.py [hunk 1]
```diff
@@ -351,7 +351,12 @@ def update(self, params: QueryParamTypes = None) -> None:  # type: ignore
 
         params = QueryParams(params)
         for param in params:
-            self[param] = params[param]
+            item, *extras = params.getlist(param)
```
Comment (tomchristie) [inline httpx/models.py]: wondering if there's an scope for refactoring for simplificity here.
Would the following also work?...

```python
items = params.getlist(param)
self._list.extend((param, item) for item in items)
self._dict[param] = items[0]
```

Comment (kcen) [reply httpx/models.py]: I don't think that will work if the params getting replaced are of a list/repeated value.

We need to call `self[param]=` at some point to trigger the search delete for other values in `QueryParams._list` via `QueryParams.__setitem__`

Comment (tomchristie) [reply httpx/models.py]: Good point!

**Comment by tomchristie** [comment]
Thanks!

**Pull request merged**